### PR TITLE
Remove etcd patch and enforce hostname requirements

### PR DESCRIPTION
**Pull request by stbenjam**
This does not appear to be required for a default emulated baremetal (VM's) dev-scripts install, as I get a clean install. 

I can't seem to get any answers about what it's use is here in 06_create_cluster.sh, so I am removing it.  This is required to make progress on removing the early exit from our installer fork.  Both https://github.com/openshift-metal3/dev-scripts/pull/407 and https://github.com/openshift-metal3/dev-scripts/issues/333 indicate to me that baremetal users are running this separate `fix_ep.sh` script during the bootstrap process, and not relying on this hack in 06_create_cluster.sh which runs after.

### CODE REVIEW

**Comment by karmab** [comment]
i think it's ok to remove it, but we should inform the end user that they need their baremetal hosts to have a specific fqdn pattern, XX-master-NUMBER, and the same for worker.
We could then also remove the need for fix_ep.sh too

**Comment by hardys** [comment]
If this works then lgtm - perhaps we should update the README to capture the fqdn requirement though?

**Comment by stbenjam** [comment]
>  i think it's ok to remove it, but we should inform the end  user that they need their baremetal hosts to have a specific fqdn  pattern, XX-master-$digit, and the same for worker. We could then also remove the need for fix_ep.sh too

@karmab What is the XX prefix in that pattern?

**Comment by mcornea** [comment]
The fqdn in master-*/worker-* format wasn't part of the initial requirements. Since it's a new user configuration that needs to be done in the external DHCP server are we sure we want to introduce it?

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/744/

**Comment by karmab** [comment]
> > i think it's ok to remove it, but we should inform the end  user that they need their baremetal hosts to have a specific fqdn  pattern, XX-master-$digit, and the same for worker. We could then also remove the need for fix_ep.sh too
> 
> @karmab What is the XX prefix in that pattern?

XX can be anything, but  i think it s bet to set it to the cluster name.

**Comment by karmab** [comment]
> The fqdn in master-_/worker-_ format wasn't part of the initial requirements. Since it's a new user configuration that needs to be done in the external DHCP server are we sure we want to introduce it?

as an end user, i d prefer to have it as a requirement rather than having it to manually patch the cluster in the middle of the install.
that said, maybe we could use an additional static pod on any of the master node to actually perform that task? @russellb @stbenjam ?

**Comment by stbenjam** [comment]
I've updated this to remove everything and updated the readme. This constraint seems to live in openshift itself, doesn't it? Can we get a proper fix there?

File: README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ canonical way to set up a metalkube cluster.
 - ideally on a bare metal host
 - run as a user with passwordless sudo access
 - get a valid pull secret (json string) from https://cloud.openshift.com/clusters/install#pull-secret
+- hostnames for masters and workers must be in the format XX-master-# (e.g. openshift-master-0), or XX-worker-# (e.g. openshift-worker-0)
```
Comment (mcornea) [inline README.md]: Will it work with FQDN, e.g: openshift-master-0.qe1.kni.lab.eng.bos.redhat.com ?

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/745/

File: README.md (hunk 1 above)
Comment (stbenjam) [reply README.md]: I believe as long as qe1.kni.lab.eng.bos.redhat.com is the $CLUSTER_DOMAIN, this is how the VM environment looks:

```console
$ hostname -f
master-0.ostest.test.metalkube.org
```

**Comment by hardys** [comment]
> The fqdn in XX-master-$digit/XX-worker-$digit format wasn't part of the initial requirements. Since it's a new user configuration that needs to be done in the external DHCP server are we sure we want to introduce it?

To enable end-to-end deployment via kni-installer we have to move away from the dev-scripts hack, so I think it's reasonable to document this as a constraint, and if it turns out to be problematic for users, we'll raise a bug/RFE to track a properly integrated solution.

**Comment by celebdor** [comment]
the name schema needs to be exactly master-0, master-1, master-2 IIRC

**Comment by stbenjam** [comment]
The VM environment uses openshift-master-0, etc as the hostname and it seems to work - or is the name schema something else?

**Comment by stbenjam** [comment]
Sorry that's the name of the VM. It does appear the hostname is master-0:

```
$ hostname -f
master-0.ostest.test.metalkube.org
```

Will follow up with an update to the README, thanks.

**Comment by derekhiggins** [comment]
I guess this means that using a BM environment we would end up lock the purpose of 3 specific hosts to be masters is this ok?

**Comment by stbenjam** [comment]
> I guess this means that using a BM environment we would end up locking the purpose of 3 specific hosts to be masters is this ok?

I would guess ultimately it's not, but it can't live in dev-scripts, where does this need to get tracked so it can be fixed properly?

**Comment by derekhiggins** [comment]
> > I guess this means that using a BM environment we would end up locking the purpose of 3 specific hosts to be masters is this ok?
> 
> I would guess ultimately it's not, but it can't live in dev-scripts, where does this need to get tracked so it can be fixed properly?

Afraid I don't know

**Comment by stbenjam** [comment]
@celebdor Could you help us figure out where a fix for this needs to get handled? Is it something we could do in the installer?

File: README.md (hunk 1 above)
Comment (karmab) [reply README.md]: it does work with XX-master-0 and so on

**Pull request merged**